src/conn/conn_sweep.c

```diff
@@ -235,21 +239,15 @@ __sweep_remove_handles(WT_SESSION_IMPL *session, time_t now)
 		dhandle_next = TAILQ_NEXT(dhandle, q);
 		if (WT_IS_METADATA(dhandle))
 			continue;
-		if (F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE | WT_DHANDLE_OPEN) ||
-		    dhandle->session_inuse != 0 || dhandle->session_ref != 0)
-			continue;
-		if (dhandle->timeofdiscard != 0 &&
-		    now <= dhandle->timeofdiscard + conn->sweep_idle_time)
+		if (!WT_DHANDLE_CAN_DISCARD(dhandle))
 			continue;
 
 		WT_WITH_HANDLE_LIST_LOCK(session,
 		    ret = __sweep_remove_one(session, dhandle));
 		if (ret == 0)
 			WT_STAT_FAST_CONN_INCR(session, dh_sweep_remove);
-		else {
+		else
```

Comment:
We should add a comment why non-zero return means we should increment the sweep reference stat (i.e. the handle was idle and by the time we grabbed the lock to remove it, someone started using it again).  Do we only expect EBUSY in that case?  If so maybe we should assert that or only increment the stat on that error?